apps/sms/index.html

```diff
@@ -148,6 +155,33 @@ <h1 data-l10n-id="messages"></h1>
           <a id="threads-settings-button" data-icon="settings" data-l10n-id="settingsButton"></a>
         </gaia-header>
         <article id="threads-container" class="view-body edit-container" data-type="list">
+          <gaia-fast-list id="threads-list" top="0" bottom="0" caching>
+            <!--<template header>
+              <gaia-sub-header data-time="${daySection}">${section}</gaia-sub-header>
+            </template>-->
+            <template item>
```

Comment:
> I don't understand what advantage this gives us. What's wrong with passing localized strings to the list to render?

Mmm, I see at least two drawbacks of current way:

* Every app should track language change event and re-localize entire data set;
* And if data item contains several localizable strings, we'll have somewhat complex code to prepare data model because of async nature of ```document.l10n.formatValue```, e.g.
```
let rawData = [{ field1, field2 }, { field1, field2 }];

let localizedItemPromises = rawData.map((rawItem) => {
    // It's very naive example, in reality there is always some logic behind localization
    // for every field.
	return Promise.all([
		document.l10n.formatValue(rawItem.field1),
		document.l10n.formatValue(rawItem.field1)
	]).then([localizedField1, localizedField2] => {
		return { localizedField1, localizedField2 };
	});
});

Promise.all(localizedItemPromises).then((localizedItems) => {
	list.setModel(localizedItems);
});

+ Do the same job on language change, edge case but still
```

Or I'm missing something? IMO with such code we're loosing some of the benefits of l10n lib and do heavy lifting on our own :/